### PR TITLE
feat: GRE tunnel support for Transit Gateway

### PR DESCRIPTION
**Pull request by kennburger**
## PR summary
Support for a Transit gateway connection of type gre_tunnel

**Fixes:** <! -- link to issue -->

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [ ] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## What is the new behavior?  
New parameters have been added for creating a Transit Gateway connection that support the creation of a GRE tunnel connection.

## Does this PR introduce a breaking change?    
- [ ] Yes
- [ ] No

### CODE REVIEW

File: transitgatewayapisv1/transit_gateway_apis_v1_integration_test.go [hunk 1]
```diff
@@ -361,6 +361,136 @@ var _ = Describe(`TransitGatewayApisV1`, func() {
...
+						Expect(*response.ID).To(Equal(os.Getenv("CONN_INSTANCE_ID_CLASSIC")))
+						break
+					}
+
+					// other than attached, see if we have reached the timeout value.  If so, exit with failure
+					if timer > 24 { // 2 min timer (24x5sec)
+						Expect(*response.Status).To(Equal("attached")) // timed out fail if status is not attached
+						break
+					} else {
+						// Still exists, wait 5 sec
```
Comment (robgolley) [inline transitgatewayapisv1/transit_gateway_apis_v1_integration_test.go:417]: 'Still exists' doesn't apply here.  Maybe just 'wait 5 sec'

File: transitgatewayapisv1/transit_gateway_apis_v1_integration_test.go [hunk 2]
```diff
@@ -508,7 +638,49 @@ var _ = Describe(`TransitGatewayApisV1`, func() {
...
+						Expect(detailedResponse.StatusCode).To(Equal(404)) // response is 404, exit success
+						break
+					}
+
+					// other than 404, see if we have reached the timeout value.  If so, exit with failure
+					if timer > 24 { // 4 min timer (24x10sec)
+						Expect(detailedResponse.StatusCode).To(Equal(404)) // timed out fail if code is not 404
+						break
+					} else {
+						// Still exists, wait 5 sec
```
Comment (robgolley) [inline transitgatewayapisv1/transit_gateway_apis_v1_integration_test.go:676]: should be 10 seconds

File: transitgatewayapisv1/transit_gateway_apis_v1_integration_test.go [hunk 3]
```diff
@@ -550,7 +722,48 @@ var _ = Describe(`TransitGatewayApisV1`, func() {
...
```
Comment (robgolley) [inline transitgatewayapisv1/transit_gateway_apis_v1_integration_test.go:760]: 10 sec, not 5

**Review by robgolley — APPROVED** [review]
lg.  Couple of nit comments

**Review by sarah-ulmer — APPROVED** [review]

**Pull request merged**